datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -396,6 +397,39 @@ fn string_to_timestamp_nanos_shim(s: &str) -> Result<i64> {
     string_to_timestamp_nanos(s).map_err(|e| e.into())
 }
 
+fn to_date_impl(args: &[ColumnarValue], name: &str) -> Result<ColumnarValue> {
+    match args.len() {
+        1 => handle::<Date32Type, _, Date32Type>(
+            args,
+            |s| {
+                string_to_timestamp_nanos_shim(s)
```

Comment:
I still think for one-arg version of this impl we should just use the arrow function to cast from string -> date. Untested but something like this:
`1 => cast_column(&args[0], &DataType::Date32, None),`